### PR TITLE
chore: remove react-scripts, use vite

### PR DESCRIPTION
**Pull request by rluders**
This PR does a code clean and updates our tools:

- Replace `react-scripts` with `vite` (faster, better, modern, less trash)
- Replace `jest` with `vitest` (faster, better fit to use with vite)
- Updates `flowbite`, `flowbite-react` and `tailwindcss`
- Removes some meaningless unit tests, to give space to better unit tests
- Improves ESLint configuration
- Improves Prettier configuration
- Prettify the code
- Moves `mock_server/` to `tools/mockserver/`
- Moves `scripts/translations.js` to `tools/translations/index.js`
- Update README with new instructions (`yarn starts` now runs the mock and the react app)
- Removes some old local components (we are using the flowbite ones)

**Out of scope**

- Fix lint-reported issues
- Fix flowbite-react styled/themed components

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -1,5 +1,11 @@
-**/*.js
-node_modules
-build
+node_modules/
+build/
+cypress/
```
Comment (arianerocha) [inline .eslintignore:3]: I think the test code used in cypress should also be verified by lint.

File: .prettierignore [hunk 2]
```diff
@@ -0,0 +1,5 @@
+node_modules/
+build/
+cypress/
```
Comment (arianerocha) [inline .prettierignore:3]: I think we can remove cypress from the ignored list

File: .prettierrc.js [hunk 3]
```diff
@@ -1,8 +1,8 @@
-{
```
Comment (arianerocha) [inline .prettierrc.js:1]: non-blocker: what's the difference between keeping the files as `.prettierrc` instead to change to `.prettierrc.js`

**Review by arianerocha — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -65,9 +65,9 @@ Create an **.env** file on root directory
 ```.sh
 SKIP_PREFLIGHT_CHECK=true
 
-REACT_APP_KEYCLOAK_URL=http://sso.sumelms.sol
-REACT_APP_KEYCLOAK_REALM=sume
-REACT_APP_KEYCLOAK_CLIENT_ID=sume-app
+VITE_APP_KEYCLOAK_URL=http://sso.sumelms.sol
+VITE_APP_KEYCLOAK_REALM=sume
+VITE_APP_KEYCLOAK_CLIENT_ID=sume-app
```
Comment (arianerocha) [inline README.md:70]: can you include the mock env here? `VITE_APP_BASE_SERVER_URL=http://localhost:9000/api`

**Review by arianerocha — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -65,9 +65,9 @@ Create an **.env** file on root directory
 ```.sh
 SKIP_PREFLIGHT_CHECK=true
 
-REACT_APP_KEYCLOAK_URL=http://sso.sumelms.sol
-REACT_APP_KEYCLOAK_REALM=sume
-REACT_APP_KEYCLOAK_CLIENT_ID=sume-app
+VITE_APP_KEYCLOAK_URL=http://sso.sumelms.sol
```
Comment (arianerocha) [inline README.md:68]: I think you have to also fix the `.env.example`

**Review by arianerocha — CHANGES_REQUESTED** [review]

File: .prettierrc.js (hunk 3 above)
Comment (rluders) [reply .prettierrc.js:1]: I think JSON format is easier to break and make mistakes. Using `.js` we do have linter support and other stuff.

**Review by arianerocha — APPROVED** [review]

**Pull request merged**